src/sage/schemes/riemann_surfaces/riemann_surface.py

```diff
@@ -4669,15 +4569,26 @@ def partner(col):
 
     N = M_curr
 
+    print("N:\n", N)
+
     if transformation:
         return N, S.inverse(), T.inverse()
 
     return N
 
-
 def reduce_riemann_matrix(M, Z):
     r"""
-    Reduce the Riemann matrix Z of the Riemann surface to Z'.
+    Reduce the Riemann matrix `Z` of the Riemann surface to `Z'` using a 
+    reduction matrix `M` following Martens' algorithm in [Mar1992]_.
+
+    The matrix `M` is a reduction matrix associated to the Riemann surface 
+    which may typically be obtained from an idempotent `e` of the associated 
+    endomorphism algebra of the Riemann surface. The row space of `e` gives 
+    rise to a corresponding reduction matrix `M` which encodes the reduction 
+    of the Riemann surface.
+
+    When `Z` is reduced to `Z'`, `Z'` represents the symplectic change of 
```

Comment:
I'm not sure what you're saying here is entirely correct. `Z'` is the Riemann matrix of a factor of the isogeny decomposition of the Jacobian associated to `Z`. I think the matrix `T` coming from `poincare_form(M)` is the symplectic change of homology basis such that the reduction encoded by `M` is just projection onto the first `2m` basis elements of the lattice.